File: src/main/java/com/kmarzecki/communicator/util/DateTimeProvider.java
package com.kmarzecki.communicator.util;

import java.time.LocalDateTime;
import java.util.Date;

/**
 * Provider for time data
 */
public interface DateTimeProvider {
    /**
     * @return current date
     */
    Date currentDate();

    /**
     * @return current datetime
     */
    LocalDateTime currentLocalDateTime();
}

File: src/main/kotlin/com/kmarzecki/trollo/util/DateTimeProvider.kt
package com.kmarzecki.trollo.util

import java.time.LocalDateTime
import java.util.*

/**
 * Provider for time data
 */
interface DateTimeProvider {
    /**
     * @return current date
     */
    fun currentDate(): Date

    /**
     * @return current datetime
     */
    fun currentLocalDateTime(): LocalDateTime?
}